code/modules/mob/living/silicon/robot/emote.dm

```diff
@@ -228,15 +228,15 @@
 			if(world.time-lastvent >= 30 SECONDS)
 				var/list/robotfarts = list("vents","shakes violently, then vents")
 				var/robofart = pick(robotfarts)
-				for(var/mob/living/M in view(0))
-					if(M != src && M.loc == src.loc)
+				for(var/mob/living/M in loc)
+					if(M != src && M.loc == src.loc && M.lying || M.resting)
```

Comment:
Operator precedence is gonna screw you over here, move the `||` into parentheses.